tests/foreman/api/test_contentviewversion.py

```diff
@@ -66,16 +70,65 @@ def test_delete_version(self):
         content_view = content_view.read()
         # Get published content-view version id
         self.assertEqual(len(content_view.version), 1)
-        # Get 'Library' life-cycle environment id
-        response = client.get(
-            entities.LifecycleEnvironment().path(),
-            auth=get_server_credentials(),
-            data={u'organization_id': org.id},
-            verify=False,
-        )
-        response.raise_for_status()
-        lc_env_id = response.json()['results'][0]['id']
+        cvv = content_view.version[0].read()
+        self.assertEqual(len(cvv.environment), 1)
         # Delete the content-view version from selected env
-        content_view.delete_from_environment(lc_env_id)
+        content_view.delete_from_environment(cvv.environment[0].id)
         # Delete the version
         content_view.version[0].delete()
+        # Make sure that content view version is really removed
+        content_view = content_view.read()
+        self.assertEqual(len(content_view.version), 0)
+
+    def test_delete_version_non_default(self):
+        """@Test: Create content view and publish and promote it to new
+        environment. After that try to disassociate content view from 'Library'
+        and one more non-default environments through 'delete_from_environment'
+        command and delete content view version from that content view.
+
+        @Assert: Content view version deleted successfully
+
+        @Feature: ContentViewVersion
+
+        """
+        org = entities.Organization().create()
+        content_view = entities.ContentView(organization=org).create()
+        # Publish content view
+        content_view.publish()
+        content_view = content_view.read()
+        self.assertEqual(len(content_view.version), 1)
+        cvv = content_view.version[0].read()
+        self.assertEqual(len(cvv.environment), 1)
+        lce = entities.LifecycleEnvironment(organization=org).create()
+        content_view.version[0].promote(lce.id)
```

Comment:
```cvv``` and ```lce``` are used only once. You can drop them to smth like:
```python
        self.assertEqual(len(content_view.version[0].read().environment), 1)
        content_view.version[0].promote(
            entities.LifecycleEnvironment(organization=org).create().id
        )
```
Not sure if that looks better though.